pkg/async/notifications/implementations/cloudevent_publisher.go

```diff
@@ -0,0 +1,94 @@
+package implementations
+
+import (
+	"context"
+	"time"
+
+	"github.com/NYTimes/gizmo/pubsub"
+	"github.com/google/uuid"
+
+	pbcloudevents "github.com/cloudevents/sdk-go/binding/format/protobuf/v2"
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	"github.com/flyteorg/flyteadmin/pkg/async/notifications/interfaces"
+
+	"github.com/flyteorg/flytestdlib/logger"
+	"github.com/flyteorg/flytestdlib/promutils"
+	"github.com/golang/protobuf/proto"
+)
+
+const (
+	cloudEventType   = "com.flyte.workflow"
```

Comment:
i think for the initial implementation we only need one or two events, but I think the structure should be there to be able to translate all events we have - all of these https://github.com/flyteorg/flyteidl/blob/master/protos/flyteidl/event/event.proto

Where is the filter currently to decide which events to translate from raw event to cloudevent?

Reply:
> Where is the filter currently to decide which events to translate from raw event to cloudevent?

here. https://github.com/flyteorg/flyteadmin/pull/355/files#diff-899da2dc4f47b8d6467d0d2a8d779f36fcb7334965153ceae018e2652e9f0ac7R36-R38